app/controllers/incomeSources/add/IncomeSourceAddedController.scala

```diff
@@ -0,0 +1,164 @@
+                                            retrieveNino: NinoPredicate,
+                                            val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                            val retrieveBtaNavBar: NavBarPredicate,
+                                            val itvcErrorHandler: ItvcErrorHandler,
+                                            incomeSourceDetailsService: IncomeSourceDetailsService,
+                                            val obligationsView: IncomeSourceAddedObligations,
+                                            nextUpdatesService: NextUpdatesService)
+                                           (implicit val appConfig: FrontendAppConfig,
+                                            implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                            implicit override val mcc: MessagesControllerComponents,
+                                            val ec: ExecutionContext,
+                                            dateService: DateServiceInterface)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching with IncomeSourcesUtils {
+
+  def show(incomeSourceId: String, incomeSourceType: IncomeSourceType): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(isAgent = false, incomeSourceId, incomeSourceType)
+  }
+
+  def showAgent(incomeSourceId: String, incomeSourceType: IncomeSourceType): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleRequest(isAgent = true, incomeSourceId, incomeSourceType)
+        }
+  }
+
+  private def getBackUrl(incomeSourceId: String, isAgent: Boolean, incomeSourceType: IncomeSourceType): String = {
+    val baseRoute = incomeSourceType match {
+      case SelfEmployment =>
+        if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.showAgent _ else
+          controllers.incomeSources.add.routes.BusinessReportingMethodController.show _
+      case UkProperty =>
+        if (isAgent) controllers.incomeSources.add.routes.UKPropertyReportingMethodController.showAgent _ else
+          controllers.incomeSources.add.routes.UKPropertyReportingMethodController.show _
+      case ForeignProperty =>
+        if (isAgent) controllers.incomeSources.add.routes.ForeignPropertyReportingMethodController.showAgent _ else
+          controllers.incomeSources.add.routes.ForeignPropertyReportingMethodController.show _
+    }
+    baseRoute(incomeSourceId).url
+  }
+
+  def getIncomeSourceData(incomeSourceType: IncomeSourceType, incomeSourceId: String)(implicit user: MtdItUser[_]): Option[(LocalDate, Option[String])] = {
```

Comment:
this method is a good candidate to be moved somewhere else away from controller:
* service layer / separate trait etc
in the controller we tend to "join" results and render final result to user